### PR TITLE
PCVL-117: Introduce unitary matrix as a component

### PR DESCRIPTION
**Pull request by ericbrts**

### CODE REVIEW

**Comment by ericbrts** [comment]
This PR is still missing differentiated renderings for symb.Unitary and phys.Unitary

File: perceval/lib/phys/circuit.py [hunk 1]
```diff
@@ -384,7 +385,50 @@ def shape(self, content, canvas: Canvas, compact: bool = False):
...
+            m //= 2
+        super().__init__(m)
+
+    def _compute_unitary(self, assign: dict = None, use_symbolic: bool = False) -> Matrix:
+        # Ignore assign and use_symbolic parameters as __init__ checked the unitary matrix is numeric
+        return self._u
+
+    def inverse(self, v=True, h=False):
+        if v:
+            self._u = np.flipud(self._u)
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/lib/phys/circuit.py]: I've added `h` support and fixed `v`. I've added corresponding unit tests to make sure Unitary.inverse() works.

[the remark this answers is not in the document]
Comment (jsenellart) [reply perceval/lib/phys/circuit.py]: cool

File: perceval/lib/phys/circuit.py [hunk 2]
```diff
@@ -384,7 +385,50 @@ def shape(self, content, canvas: Canvas, compact: bool = False):
...
+        self.width = m
+        self._supports_polarization = use_polarization
+        if use_polarization:
+            assert m % 2 == 0, "Polarization matrix should have an even number of rows/col"
+            m //= 2
+        super().__init__(m)
+
+    def _compute_unitary(self, assign: dict = None, use_symbolic: bool = False) -> Matrix:
+        # Ignore assign and use_symbolic parameters as __init__ checked the unitary matrix is numeric
+        return self._u
```
[the remark this answers is not in the document]
Comment (jsenellart) [reply perceval/lib/phys/circuit.py:410]: ok perfect!

**Pull request merged**